bundles/edu.kit.kastel.sdq.eclipse.grading.core/src/edu/kit/kastel/sdq/eclipse/grading/core/ExerciseArtemisController.java

```diff
@@ -0,0 +1,138 @@
+import edu.kit.kastel.sdq.eclipse.grading.api.artemis.mapping.ISubmission;
+import edu.kit.kastel.sdq.eclipse.grading.api.controller.AbstractController;
+import edu.kit.kastel.sdq.eclipse.grading.api.controller.IExerciseArtemisController;
+import edu.kit.kastel.sdq.eclipse.grading.client.git.GitException;
+import edu.kit.kastel.sdq.eclipse.grading.client.git.GitHandler;
+import edu.kit.kastel.sdq.eclipse.grading.core.artemis.WorkspaceUtil;
+
+public class ExerciseArtemisController extends AbstractController implements IExerciseArtemisController {
+	private String username;
+	private String password;
+
+	public ExerciseArtemisController(String username, String password) {
+		this.username = username;
+		this.password = password;
+	}
+
+	@Override
+	public void loadExerciseInWorkspaceForStudent(ICourse course, IExercise exercise,
+			IProjectFileNamingStrategy projectNaming, String repoUrl) throws ArtemisClientException {
+		final File eclipseWorkspaceRoot = ResourcesPlugin.getWorkspace().getRoot().getLocation().toFile();
+
+		// abort if directory already exists.
+		this.existsAndThrow(projectNaming.getProjectFileInWorkspace(eclipseWorkspaceRoot, exercise, null));
+
+		try {
+			this.downloadExercise(exercise, eclipseWorkspaceRoot, repoUrl, projectNaming);
+		} catch (ArtemisClientException e) {
+			throw new ArtemisClientException("Error, can not download exercise.", e);
+		}
+		try {
+			WorkspaceUtil.createEclipseProject(
+					projectNaming.getProjectFileInWorkspace(eclipseWorkspaceRoot, exercise, null));
+		} catch (CoreException e) {
+			throw new ArtemisClientException("Error, can not create project.", e);
+		}
+	}
+
+	@Override
+	public void downloadExerciseAndSubmission(IExercise exercise, ISubmission submission, File dir,
+			IProjectFileNamingStrategy namingStrategy) throws ArtemisClientException {
+		final File projectDirectory = namingStrategy.getProjectFileInWorkspace(dir, exercise, submission);
+		this.existsAndThrow(projectDirectory);
+		try {
+			if (projectDirectory.exists()) {
+				throw new ArtemisClientException(
+						"Could not clone project " + projectDirectory.getName() + ", " + "directory already exists!");
+			}
+
+			// Download test repository
+			GitHandler.cloneRepo(projectDirectory, exercise.getTestRepositoryUrl(), Constants.MASTER_BRANCH_NAME);
+			// download submission inside the exercise project directory
+			GitHandler.cloneRepo(namingStrategy.getAssignmentFileInProjectDirectory(projectDirectory), //
+					submission.getRepositoryUrl(), Constants.MASTER_BRANCH_NAME);
+		} catch (GitException e) {
+			throw new ArtemisClientException("Unable to download exercise and submission: " + e.getMessage(), e);
+		}
+
+	}
+
+	public void downloadExercise(IExercise exercise, File dir, String repoUrl,
+			IProjectFileNamingStrategy namingStrategy) throws ArtemisClientException {
+		final File projectDirectory = namingStrategy.getProjectFileInWorkspace(dir, exercise, null);
+		try {
+			if (projectDirectory.exists()) {
```

Comment:
Is this code also used for grading?
If so this might break the functionality to continue an assessment bo cloning it again. 
(This might be necessarry, if the tool freezes and must be restarted.) 

However this is not a fatal issue but should definetly be checked (and then maybe get fixed in the future)